internal/appsec/waf.go

```diff
@@ -195,8 +230,19 @@ func newGRPCWAFEventListener(handle *waf.Handle, _ []string, timeout time.Durati
 			mu.Unlock()
 		}))
 		op.On(grpcsec.OnHandlerOperationFinish(func(op *grpcsec.HandlerOperation, _ grpcsec.HandlerOperationRes) {
+			op.AddTag(wafDurationTag, float64(wafRunDuration/1e3))
+			op.AddTag(wafDurationExtTag, float64(wafBindingsRunDuration/1e3))
+
+			metricsOnce.Do(func() {
+				rInfo := handle.RulesetInfo()
+				op.AddTag(eventRulesVersionTag, rInfo.Version)
+				op.AddTag(eventRulesErrorsTag, rInfo.Errors)
+				op.AddTag(eventRulesLoadedTag, float64(rInfo.Loaded))
+				op.AddTag(eventRulesFailedTag, float64(rInfo.Failed))
+				op.AddTag(ext.ManualKeep, true)
```

Comment:
Since this code is 100% the same in the HTTP listener too, can you please put it in a common in a new helper function such as `addRulesetInfoTags()`?